Explicitly ask git branch not to use colors

vhf opened the pull request:
This patch explicitly tells `git branch` not to use colors. When git is configured to display the current branch with colors, publish-please fails:

`~/.gitconfig`:
```
[color "branch"]
  current = yellow reverse
```
Result:
![pp](https://cloud.githubusercontent.com/assets/2022803/15622166/c3cbb638-2466-11e6-9392-6949b9823f4a.jpg)

I didn't include a test case because I don't know how to test this.

inikulin commented:
Nice find, thanks!

> I didn't include a test case because I don't know how to test this.

We can write `.gitconfig` to testing repo on test [setup](https://github.com/inikulin/publish-please/blob/master/test/test.js#L46). Also, it would be nice to enable all colors for all commands that supports coloring:
```sh
color.branch
color.diff
color.interactive
color.status
```
to see if it will break something else.

vhf commented:
Ok cool, thanks for the hint I'll add tests and take care of colors wherever it may break this project.

inikulin commented:
@vhf Awesome, thanks!

inikulin commented:
@vhf Any update on this?

vhf commented:
Yeah definitely! I spent some time on it last week but didn't manage to create tests for it. It seems the local .gitconfig created in my test isn't considered by git, but it could also well be that my global gitconfig overrides it.

I'll push my WIP-branch and you can take a look and perhaps help me with it.

vhf commented:
Yeah so as I explained through commit messages, I added a `.gitconfig` file adding colors everywhere for the tests. Unfortunately the tests still pass.

I'll squash this nicely once we find a solution.

inikulin commented on test/test.js:
```diff
@@ -37,12 +37,17 @@ function getTestOptions (settings) {
 }
 
+function writeConfig (from, to) {
+    writeFile(`${to}/.git/config`, readFile(`${to}/.git/config`) + readFile(`${from}/gitconfig`));
```
Have you tried writing `.gitconfig` into the root of the repo?

vhf replied:
Nope. Will do tomorrow. :)

vhf replied:
This doesn't seem to work. I thought about adding some env variables (`GIT_CONFIG_NOGLOBAL=1 GIT_CONFIG_NOSYSTEM=1`) to disable using the global git configs, but `cp-sugar` doesn't handle these.

`exec('HELLO=1337 echo $HELLO').then(x => console.log(x));`

inikulin replied:
Strange, local `.gitconfig` works perfectly for me. Are you sure it is correctly written to the testing repo? Have you tried putting it to the testing repo manually and running `git branch`?

inikulin replied:
OK, it appears it doesn't work for me. But seems like it works if you run the following in testing repo dir:
```js
function colorGitOutput () {
    var gitColorCommands = [
        'git config color.branch.current red',
        'git config color.branch.local red',
        'git config color.branch.remote red',
        'git config color.diff.meta red',
        'git config color.diff.frag red',
        'git config color.diff.old red',
        'git config color.diff.new red',
        'git config color.status.added red',
        'git config color.status.changed red',
        'git config color.status.untracked red'
    ];

    return Promise.all(gitColorCommands.map(s => exec(s)));
}
```

inikulin replied:
ping @vhf

vhf commented:
Thanks for the reminder and your insights @inikulin . It's fixed. :)

I switched to blue because it's much nicer when it fails 👯 

![screen shot 2016-08-04 at 16 10 10](https://cloud.githubusercontent.com/assets/2022803/17404992/1717e0fa-5a5e-11e6-83ba-6ba7a0cf7a12.png)

And I added another `sed` in `after()` to remove this config. Otherwise running the tests and then running `git diff` while developing keeps showing everything in blue.

inikulin commented on test/test.js:
```diff
@@ -37,20 +37,40 @@ function getTestOptions (settings) {
 
 after(() => {
     process.chdir('../');
+    exec('sed -i\'\' \'/= blue/d\' .git/config');
```
Since exec is async, it should be:
```js
return exec('sed -i\'\' \'/= blue/d\' .git/config')
      .then(() => del('testing-repo'));
```

Otherwise process may terminate before exec finished (mocha forces process termination)

inikulin commented:
Awesome! Overall LGTM, just a little minor issue left.

vhf commented:
Fixed, thanks 👍

inikulin commented:
Great, thank you!

inikulin commented:
Published as v2.2.0
